LongestPole/FormatTable.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2024 Bolton Bailey. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bolton Bailey
+-/
+import Mathlib.Data.String.Defs
+
+
+/-!
+# Format Table
+
+This file provides a simple function for formatting a 2d array of strings into a markdown-compliant table.
+-/
+
+inductive Alignment where
+  | left
+  | right
+  | center
+deriving Inhabited, BEq
+
+/--
+Takes a 2d array of string data and renders it into a table.
+ -/
+def formatTable (headers : Array String) (table : Array (Array String))
+    (alignments : Option (Array Alignment) := none) :
+    String := Id.run do
+  -- If no alignments are provided, default to left alignment for all columns
+  let alignments := alignments.getD (Array.mkArray headers.size Alignment.left)
+  -- Get the maximum widths of each column
+  let mut widths := headers.map (·.length)
+  for row in table do
+    for i in [0:widths.size] do
+      widths := widths.set! i (max widths[i]! ((row[i]?.map (·.length)).getD 0))
+  -- Pad each cell with spaces to match the column width
+  let paddedHeaders := headers.mapIdx fun i h => h.rightpad widths[i]!
+  let paddedTable := table.map fun row => row.mapIdx fun i cell =>
+    if alignments[i]! == Alignment.left then cell.rightpad widths[i]!
```

Comment:
This is a matter of taste, but: what do you think about using match here?
```
  match alignments[i] with
    | Alignments.left => cell.rightpad widths[i]!
    | Alignments.right => cell.leftpad widths[i]!
    | Alignments.center =>
            String.replicate ((widths[i]! - cell.length) / 2) ' '
            ++ cell
            ++ String.replicate ((widths[i]! - cell.length + 1) / 2) ' '

Reply:
Actually: if you'd like to refactor this even more, you could also extract a function
`def align (str : String) (length : Nat) (alignment : Alignment)`.

Update: Originally I thought you'd need the same logic below. Since the code below is slightly different, I'll leave it up to you to decide whether you want to do this.